### DISCUSSION TITLE
Should notebooks be tested within the CI pipeline, so there are kept up to date?

### DISCUSSION THREAD

**Discussion by psomhorst** [EIT-ALIVE/eitprocessing#332] [Ideas]
Inspired by https://github.com/EIT-ALIVE/eitprocessing/pull/325#issue-2623948683. 

The repository contains a number of notebooks that can function as examples on how to use the repository. Many/most/all (?) of these are not maintained and outdated. I have refrained from putting effort into them, because there is a chance they will be outdated soon-ish anyway.

I have a proposition that would increase the quality and usefulness of these notebooks.

1. We make a GitHub actions that runs the notebooks when a PR is created. This ensures that the PR does not break the functioning of the notebooks.
2. We make an example notebook for all/most features of the repo. These can be taken from existing projects or purpose-built. 
3. We add these examples to the documentation as basic and intermediate/advanced examples. 

This issue relies on #330 to be completed first.

What do you think, @JulietteFrancovich @DaniBodor?

**Comment by DaniBodor** [comment]
1. Yes, I think it would be good to add such an action. We did this in another project as well, because there it had happened once or twice that the notebook broke, while the tests all still succeeded. In theory, that should only happen if there's a major version update (minor/patch should always have backwards compatibility), so it can also serve as an indicator for major release updates.
2. I think this is unit/integration tests are for, so not sure we need to run the majority of the package inside a notebook as well.
3. If it's for this purpose, then such a notebook is nice, but I still wouldn't run a large notebook on every PR (maybe only in the release action?)
